### ISSUE TITLE
Error upgrading to v0.2.172

### ISSUE THREAD

**Issue by LisaHagenau** [miso-lims/miso-lims#1960]
Hello,
I have a problem upgrading to Version 0.2.172. Migrating the database seems to work fine, this is the output from Flyway:

```
Flyway 3.2.1 by Boxfuse

Database: jdbc:mysql://localhost:3306/miso (MySQL 5.7)
Validated 115 migrations (execution time 00:00.186s)
Executing SQL callback: beforeMigrate
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
Current version of schema `miso`: 0800
WARNING: outOfOrder mode is active. Migration of schema `miso` may not be reproducible.
Migrating schema `miso` to version 0810 - auto main
Migrating schema `miso` to version 0820 - auto main
Successfully applied 2 migrations to schema `miso` (execution time 00:00.622s).
Executing SQL callback: afterMigrate
WARNING: DB: FUNCTION miso.nextval does not exist (SQL State: null - Error Code: 1305)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: PROCEDURE miso.moveBoxItem does not exist (SQL State: null - Error Code: 1305)
WARNING: DB: PROCEDURE miso.removeBoxItem does not exist (SQL State: null - Error Code: 1305)
WARNING: DB: Unknown table 'miso.BoxableView' (SQL State: 42S02 - Error Code: 1051)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Trigger does not exist (SQL State: null - Error Code: 1360)
WARNING: DB: Unknown table 'miso.CompletedPartitions' (SQL State: 42S02 - Error Code: 1051)
```

However, after restarting tomcat, I get a HTTP Status 404 – Not Found when I am trying to access the web interface. Here is the output from the catalina and localhost logs in /opt/tomcat/logs:

```
31-May-2019 10:57:11.027 INFO [ContainerBackgroundProcessor[StandardEngine[Catalina]]] org.apache.catalina.startup.HostConfig.undeploy Undeploying context []
31-May-2019 10:57:11.162 WARNING [ContainerBackgroundProcessor[StandardEngine[Catalina]]] org.apache.catalina.loader.WebappClassLoaderBase.clearReferencesThreads The web application [ROOT] appears to have started a thread named [MySQL Statement Cancellation Timer] but has failed to stop it. This is very likely to create a memory leak. Stack trace of thread:
 java.lang.Object.wait(Native Method)
 java.lang.Object.wait(Object.java:502)
 java.util.TimerThread.mainLoop(Timer.java:526)
 java.util.TimerThread.run(Timer.java:505)
31-May-2019 10:57:11.180 INFO [ContainerBackgroundProcessor[StandardEngine[Catalina]]] org.apache.catalina.startup.HostConfig.deployDescriptor Deploying deployment descriptor [/opt/tomcat/conf/Catalina/localhost/ROOT.xml]
31-May-2019 10:57:11.188 WARNING [ContainerBackgroundProcessor[StandardEngine[Catalina]]] org.apache.catalina.startup.HostConfig.deployDescriptor The path attribute with value [/ROOT] in deployment descriptor [/opt/tomcat/conf/Catalina/localhost/ROOT.xml] has been ignored
31-May-2019 10:57:11.192 WARNING [ContainerBackgroundProcessor[StandardEngine[Catalina]]] org.apache.catalina.startup.HostConfig.deployDescriptor A docBase [/opt/tomcat/webapps/ROOT] inside the host appBase has been specified, and will be ignored
31-May-2019 10:57:37.554 INFO [ContainerBackgroundProcessor[StandardEngine[Catalina]]] org.apache.catalina.startup.HostConfig.deployDescriptor Deployment of deployment descriptor [/opt/tomcat/conf/Catalina/localhost/ROOT.xml] has finished in [26,374] ms
31-May-2019 11:03:56.590 INFO [main] org.apache.catalina.core.StandardServer.await A valid shutdown command was received via the shutdown port. Stopping the Server instance.
31-May-2019 11:03:56.590 INFO [main] org.apache.coyote.AbstractProtocol.pause Pausing ProtocolHandler ["http-nio-8080"]
31-May-2019 11:03:56.598 INFO [main] org.apache.coyote.AbstractProtocol.pause Pausing ProtocolHandler ["ajp-nio-8009"]
31-May-2019 11:03:56.599 INFO [main] org.apache.catalina.core.StandardService.stopInternal Stopping service [Catalina]
31-May-2019 11:03:56.678 INFO [main] org.apache.coyote.AbstractProtocol.stop Stopping ProtocolHandler ["http-nio-8080"]
31-May-2019 11:03:56.681 INFO [main] org.apache.coyote.AbstractProtocol.stop Stopping ProtocolHandler ["ajp-nio-8009"]
31-May-2019 11:03:56.682 INFO [main] org.apache.coyote.AbstractProtocol.destroy Destroying ProtocolHandler ["http-nio-8080"]
31-May-2019 11:03:56.683 INFO [main] org.apache.coyote.AbstractProtocol.destroy Destroying ProtocolHandler ["ajp-nio-8009"]
31-May-2019 11:04:44.768 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server version:        Apache Tomcat/9.0.8
31-May-2019 11:04:44.771 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server built:          Apr 27 2018 19:32:00 UTC
31-May-2019 11:04:44.771 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server number:         9.0.8.0
31-May-2019 11:04:44.772 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log OS Name:               Linux
31-May-2019 11:04:44.772 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log OS Version:            4.15.0-39-generic
31-May-2019 11:04:44.773 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Architecture:          amd64
31-May-2019 11:04:44.773 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Java Home:             /usr/lib/jvm/java-8-openjdk-amd64/jre
31-May-2019 11:04:44.773 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log JVM Version:           1.8.0_212-8u212-b03-0ubuntu1.18.04.1-b03
31-May-2019 11:04:44.774 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log JVM Vendor:            Oracle Corporation
31-May-2019 11:04:44.774 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log CATALINA_BASE:         /opt/tomcat
31-May-2019 11:04:44.774 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log CATALINA_HOME:         /opt/tomcat
31-May-2019 11:04:44.775 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.util.logging.config.file=/opt/tomcat/conf/logging.properties
31-May-2019 11:04:44.777 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.util.logging.manager=org.apache.juli.ClassLoaderLogManager
31-May-2019 11:04:44.777 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dsecurity.method=jdbc
31-May-2019 11:04:44.778 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Xmx768M
31-May-2019 11:04:44.778 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djdk.tls.ephemeralDHKeySize=2048
31-May-2019 11:04:44.778 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.protocol.handler.pkgs=org.apache.catalina.webresources
31-May-2019 11:04:44.779 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dorg.apache.catalina.security.SecurityListener.UMASK=0027
31-May-2019 11:04:44.779 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dignore.endorsed.dirs=
31-May-2019 11:04:44.779 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dcatalina.base=/opt/tomcat
31-May-2019 11:04:44.779 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dcatalina.home=/opt/tomcat
31-May-2019 11:04:44.780 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.io.tmpdir=/opt/tomcat/temp
31-May-2019 11:04:44.781 INFO [main] org.apache.catalina.core.AprLifecycleListener.lifecycleEvent Loaded APR based Apache Tomcat Native library [1.2.21] using APR version [1.6.3].
31-May-2019 11:04:44.785 INFO [main] org.apache.catalina.core.AprLifecycleListener.lifecycleEvent APR capabilities: IPv6 [true], sendfile [true], accept filters [false], random [true].
31-May-2019 11:04:44.785 INFO [main] org.apache.catalina.core.AprLifecycleListener.lifecycleEvent APR/OpenSSL configuration: useAprConnector [false], useOpenSSL [true]
31-May-2019 11:04:44.789 INFO [main] org.apache.catalina.core.AprLifecycleListener.initializeSSL OpenSSL successfully initialized [OpenSSL 1.1.0g  2 Nov 2017]
31-May-2019 11:04:44.974 INFO [main] org.apache.coyote.AbstractProtocol.init Initializing ProtocolHandler ["http-nio-8080"]
31-May-2019 11:04:45.006 INFO [main] org.apache.tomcat.util.net.NioSelectorPool.getSharedSelector Using a shared selector for servlet write/read
31-May-2019 11:04:45.018 INFO [main] org.apache.coyote.AbstractProtocol.init Initializing ProtocolHandler ["ajp-nio-8009"]
31-May-2019 11:04:45.023 INFO [main] org.apache.tomcat.util.net.NioSelectorPool.getSharedSelector Using a shared selector for servlet write/read
31-May-2019 11:04:45.029 INFO [main] org.apache.catalina.startup.Catalina.load Initialization processed in 1028 ms
31-May-2019 11:04:45.091 INFO [main] org.apache.catalina.core.StandardService.startInternal Starting service [Catalina]
31-May-2019 11:04:45.092 INFO [main] org.apache.catalina.core.StandardEngine.startInternal Starting Servlet Engine: Apache Tomcat/9.0.8
31-May-2019 11:04:45.113 INFO [main] org.apache.catalina.startup.HostConfig.deployDescriptor Deploying deployment descriptor [/opt/tomcat/conf/Catalina/localhost/ROOT.xml]
31-May-2019 11:04:45.156 WARNING [main] org.apache.catalina.startup.HostConfig.deployDescriptor The path attribute with value [/ROOT] in deployment descriptor [/opt/tomcat/conf/Catalina/localhost/ROOT.xml] has been ignored
31-May-2019 11:04:45.177 WARNING [main] org.apache.catalina.startup.HostConfig.deployDescriptor A docBase [/opt/tomcat/webapps/ROOT] inside the host appBase has been specified, and will be ignored
31-May-2019 11:04:50.876 WARNING [main] org.apache.tomcat.dbcp.dbcp2.BasicDataSourceFactory.getObjectInstance Name = MISODB Property maxActive is not used in DBCP2, use maxTotal instead. maxTotal default value is 8. You have set value of "100" for "maxActive" property, which is being ignored.
31-May-2019 11:04:50.882 WARNING [main] org.apache.tomcat.dbcp.dbcp2.BasicDataSourceFactory.getObjectInstance Name = MISODB Property removeAbandoned is not used in DBCP2, use one or both of removeAbandonedOnBorrow or removeAbandonedOnMaintenance instead. Both have default value set to false. You have set value of "true" for "removeAbandoned" property, which is being ignored.
31-May-2019 11:04:50.882 WARNING [main] org.apache.tomcat.dbcp.dbcp2.BasicDataSourceFactory.getObjectInstance Name = MISODB Property maxWait is not used in DBCP2 , use maxWaitMillis instead. maxWaitMillis default value is -1. You have set value of "1000" for "maxWait" property, which is being ignored.
31-May-2019 11:04:51.713 INFO [main] org.apache.jasper.servlet.TldScanner.scanJars At least one JAR was scanned for TLDs yet contained no TLDs. Enable debug logging for this logger for a complete list of JARs that were scanned but no TLDs were found in them. Skipping unneeded JARs during scanning can improve startup time and JSP compilation time.
31-May-2019 11:05:13.486 SEVERE [main] org.apache.catalina.core.StandardContext.startInternal One or more listeners failed to start. Full details will be found in the appropriate container log file
31-May-2019 11:05:13.493 SEVERE [main] org.apache.catalina.core.StandardContext.startInternal Context [] startup failed due to previous errors
31-May-2019 11:05:13.581 WARNING [main] org.apache.catalina.loader.WebappClassLoaderBase.clearReferencesThreads The web application [ROOT] appears to have started a thread named [MySQL Statement Cancellation Timer] but has failed to stop it. This is very likely to create a memory leak. Stack trace of thread:
 java.lang.Object.wait(Native Method)
 java.lang.Object.wait(Object.java:502)
 java.util.TimerThread.mainLoop(Timer.java:526)
 java.util.TimerThread.run(Timer.java:505)
31-May-2019 11:05:13.597 INFO [main] org.apache.catalina.startup.HostConfig.deployDescriptor Deployment of deployment descriptor [/opt/tomcat/conf/Catalina/localhost/ROOT.xml] has finished in [28,483] ms
31-May-2019 11:05:13.599 INFO [main] org.apache.coyote.AbstractProtocol.start Starting ProtocolHandler ["http-nio-8080"]
31-May-2019 11:05:13.633 INFO [main] org.apache.coyote.AbstractProtocol.start Starting ProtocolHandler ["ajp-nio-8009"]
31-May-2019 11:05:13.652 INFO [main] org.apache.catalina.startup.Catalina.start Server startup in 28622 ms
```
```
31-May-2019 10:57:11.039 INFO [ContainerBackgroundProcessor[StandardEngine[Catalina]]] org.apache.catalina.core.ApplicationContext.log Destroying Spring FrameworkServlet 'miso'
31-May-2019 10:57:11.049 INFO [ContainerBackgroundProcessor[StandardEngine[Catalina]]] org.apache.catalina.core.ApplicationContext.log Closing Spring root WebApplicationContext
31-May-2019 10:57:11.126 INFO [ContainerBackgroundProcessor[StandardEngine[Catalina]]] org.apache.catalina.core.ApplicationContext.log Shutting down log4j
31-May-2019 10:57:16.433 INFO [ContainerBackgroundProcessor[StandardEngine[Catalina]]] org.apache.catalina.core.ApplicationContext.log No Spring WebApplicationInitializer types detected on classpath
31-May-2019 10:57:16.527 INFO [ContainerBackgroundProcessor[StandardEngine[Catalina]]] org.apache.catalina.core.ApplicationContext.log Set web app root system property: 'webapp.root' = [/opt/tomcat/webapps/ROOT/]
31-May-2019 10:57:16.549 INFO [ContainerBackgroundProcessor[StandardEngine[Catalina]]] org.apache.catalina.core.ApplicationContext.log Initializing Spring root WebApplicationContext
31-May-2019 10:57:37.457 INFO [ContainerBackgroundProcessor[StandardEngine[Catalina]]] org.apache.catalina.core.ApplicationContext.log Initializing Spring FrameworkServlet 'miso'
31-May-2019 11:03:56.602 INFO [main] org.apache.catalina.core.ApplicationContext.log Destroying Spring FrameworkServlet 'miso'
31-May-2019 11:03:56.609 INFO [main] org.apache.catalina.core.ApplicationContext.log Closing Spring root WebApplicationContext
31-May-2019 11:03:56.652 INFO [main] org.apache.catalina.core.ApplicationContext.log Shutting down log4j
31-May-2019 11:04:51.864 INFO [main] org.apache.catalina.core.ApplicationContext.log No Spring WebApplicationInitializer types detected on classpath
31-May-2019 11:04:51.997 INFO [main] org.apache.catalina.core.ApplicationContext.log Set web app root system property: 'webapp.root' = [/opt/tomcat/webapps/ROOT/]
31-May-2019 11:04:52.015 INFO [main] org.apache.catalina.core.ApplicationContext.log Initializing Spring root WebApplicationContext
31-May-2019 11:05:13.484 SEVERE [main] org.apache.catalina.core.StandardContext.listenerStart Exception sending context initialized event to listener instance of class [uk.ac.bbsrc.tgac.miso.webapp.context.MisoAppListener]
 java.lang.IllegalStateException: Failed to link freezer maps directory
	at uk.ac.bbsrc.tgac.miso.webapp.context.MisoAppListener.linkMapsDir(MisoAppListener.java:142)
	at uk.ac.bbsrc.tgac.miso.webapp.context.MisoAppListener.contextInitialized(MisoAppListener.java:113)
	at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:4627)
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5091)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:742)
	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:718)
	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:703)
	at org.apache.catalina.startup.HostConfig.deployDescriptor(HostConfig.java:630)
	at org.apache.catalina.startup.HostConfig$DeployDescriptor.run(HostConfig.java:1840)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:112)
	at org.apache.catalina.startup.HostConfig.deployDescriptors(HostConfig.java:525)
	at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:424)
	at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1585)
	at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:308)
	at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:123)
	at org.apache.catalina.util.LifecycleBase.setStateInternal(LifecycleBase.java:424)
	at org.apache.catalina.util.LifecycleBase.setState(LifecycleBase.java:367)
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:966)
	at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:839)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1427)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1417)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:134)
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:943)
	at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:258)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.StandardService.startInternal(StandardService.java:422)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:770)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.startup.Catalina.start(Catalina.java:682)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:350)
	at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:492)
Caused by: java.nio.file.AccessDeniedException: /opt/tomcat/webapps/ROOT/freezermaps
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:84)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixFileSystemProvider.createSymbolicLink(UnixFileSystemProvider.java:457)
	at java.nio.file.Files.createSymbolicLink(Files.java:1043)
	at uk.ac.bbsrc.tgac.miso.webapp.context.MisoAppListener.linkMapsDir(MisoAppListener.java:140)
	... 42 more

31-May-2019 11:05:13.505 INFO [main] org.apache.catalina.core.ApplicationContext.log Closing Spring root WebApplicationContext
31-May-2019 11:05:13.561 INFO [main] org.apache.catalina.core.ApplicationContext.log Shutting down log4j
```

Even though we don't use freezer maps, I have added `<Resources allowLinking="true"/>` to the ROOT.xml file as suggested in the upgrade notes. 

I hope you guys can help me.
Thanks

**Comment by djcooke** [comment]
Thanks for reporting this. Can you provide a bit of additional information to help us debug?

1. Does your `miso.fileStorageDirectory` setting in `miso.properties` end in a slash (/)?
2. Does the tomcat user have rwx (read+write+execute) access to that directory?
3. Has the `freezermaps` directory been created within your storage directory?
4. If the `freezermaps` directory does exist, did you create it yourself, or was it created automatically by MISO?
5. Does the tomcat user have rwx access to the `freezermaps` directory?
6. Are you able to successfully upload an attachment to a project, sample, library, pool, run, or service record?

**Comment by LisaHagenau** [comment]
Let's see...
1. yes
2. the user tomcat has rwx access (same user that owns $CATALINA_HOME), but I do the upgrades as a different user with sudo rights
3. yes
4. created by MISO
5. see 2.
6. I can't currently access MISO and can't remember having tried before, but I can check if it works with a previous version

I have noticed that `/storage/miso` is a symbolic link pointing somewhere else. Could that be a problem?

Thanks!

**Comment by LisaHagenau** [comment]
Uploading attachments works in version 0.2.169.

**Comment by djcooke** [comment]
I just tried making `/storage/miso` a symlink on my local deployment and had no issues.

The exception says access denied when trying to create the symlink at `/opt/tomcat/webapps/ROOT/freezermaps`. Can you verify that `/opt/tomcat/webapps/ROOT` directory exists, and that the `tomcat` user has rwx access to it? Assuming you just copied the WAR file into the `webaps` dir and let Tomcat extract it normally, this should be the case. If you extracted it manually, the permissions may be wrong.

We have an upcoming priority to change the way freezer maps are handled again, so that they can be managed from within MISO. You could stay on v0.2.169 until that change goes in if you'd like. v0.2.180 gets created tomorrow, so I'm probably not going to get it in before then. It should be in v0.2.181, which gets created June 20 and made into an official release July 4. If you'd prefer to upgrade sooner, I'm happy to continue troubleshooting though.

**Comment by LisaHagenau** [comment]
The permissions were wrong for the `/opt/tomcat/webapps/ROOT` directory. Updating works now with some minor problems: 
- the institute logo is not displayed anymore, instead the name appears on the upper right
- the table `miso.persistent_logins` gets lost during the upgrade, giving the following error code when logging out

```
PreparedStatementCallback; bad SQL grammar [delete from persistent_logins where username = ?]; nested exception is com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Table 'miso.persistent_logins' doesn't exist

    org.springframework.jdbc.support.SQLErrorCodeSQLExceptionTranslator.doTranslate(SQLErrorCodeSQLExceptionTranslator.java:231)
    org.springframework.jdbc.support.AbstractFallbackSQLExceptionTranslator.translate(AbstractFallbackSQLExceptionTranslator.java:73)
    org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:655)
    org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:876)
    org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:937)
    org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:947)
    org.springframework.security.web.authentication.rememberme.JdbcTokenRepositoryImpl.removeUserTokens(JdbcTokenRepositoryImpl.java:114)
    org.springframework.security.web.authentication.rememberme.PersistentTokenBasedRememberMeServices.logout(PersistentTokenBasedRememberMeServices.java:181)
    org.springframework.security.web.authentication.logout.CompositeLogoutHandler.logout(CompositeLogoutHandler.java:54)
    org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:109)
    org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
    org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:66)
    org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
    org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
    org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:56)
    org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
    org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
    org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:105)
    org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
    org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:214)
    org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:177)
    org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:347)
    org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:263)
    org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
    org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
    org.apache.catalina.filters.ExpiresFilter.doFilter(ExpiresFilter.java:1230)
    org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
    org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
    uk.ac.bbsrc.tgac.miso.webapp.context.ExposeRequestUrlFilter.doFilter(ExposeRequestUrlFilter.java:94)
    org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:347)
    org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:263)
    org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
    org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
    org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:197)
    org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
    org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
    org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
    org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:199)
    org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
    org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:494)
    org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
    org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
    org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:651)
    org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
    org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
    org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:412)
    org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
    org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:754)
    org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1385)
    org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
    java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
    java.lang.Thread.run(Thread.java:748)
```
I downgraded back to v0.2.169 and had no problems there.

**Comment by djcooke** [comment]
We've removed the institute logo from the header as part of a gradual redesign. v0.2.175 changes the header again, making it much smaller to minimize wasted space.

The error on logout is a known issue that is annoying, but doesn't do any harm, as the user is still logged out properly. The problem is fixed in v0.2.173. It is also possible to fix in v0.2.172 by running the following SQL on your database:

```
CREATE TABLE `persistent_logins` (
  `username` varchar(64) NOT NULL,
  `series` varchar(64) NOT NULL,
  `token` varchar(64) NOT NULL,
  `last_used` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
  PRIMARY KEY (`series`)
) ENGINE = InnoDB ROW_FORMAT = DEFAULT;
```

With that said, persistent logins (the "Stay signed in" checkbox on login) have not been working for a long time, but will finally be fixed in v0.2.180.

**Comment by LisaHagenau** [comment]
Thank you, I'll just keep upgrading then :)
